packages/nimble-components/src/banner/tests/banner.stories.ts

```diff
@@ -0,0 +1,130 @@
+import type { Meta, StoryObj } from '@storybook/html';
+import { withXD } from 'storybook-addon-xd-designs';
+import { html, when } from '@microsoft/fast-element';
+import { createUserSelectedThemeStory } from '../../utilities/tests/storybook';
+import '../../all-components';
+import { BannerSeverity } from '../types';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const ActionType = {
+    none: 'none',
+    buttonGhost: 'button (ghost)',
+    buttonOutline: 'button (outline)',
+    anchor: 'anchor'
+} as const;
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+type ActionType = typeof ActionType[keyof typeof ActionType];
+
+interface BannerArgs {
+    open: boolean;
+    title: string;
+    text: string;
+    severity: BannerSeverity;
+    action: ActionType;
+    preventDismiss: boolean;
+    titleHidden: boolean;
+    dismissButtonLabel: string;
+    toggle: unknown;
+}
+
+const overviewText = `The banner is a component used to display a persistent notification to a user.
+
+Banner messages should be limited to approximately three lines of text under normal display size at the intended location. Multiple banners may be stacked vertically in order of age, with the newest at the top. Stacked banners should be spaced 1px apart.
```

Comment:
Feedback from demo: let's add usage context that describes the expected layout of banners within the page (spanning the full width of their parent).

Reply:
I added a note to the overview.